modules/reindex/src/main/java/org/elasticsearch/reindex/Reindexer.java

```diff
@@ -122,7 +129,10 @@ public void execute(BulkByScrollTask task, ReindexRequest request, Client bulkCl
                     clusterService.state(),
                     reindexSslConfig,
                     request,
-                    listener
+                    ActionListener.runAfter(listener, () -> {
+                        long elapsedTime = TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime);
+                        if (reindexMetrics != null) reindexMetrics.recordTookTime(elapsedTime);
```

Comment:
We normally prefer the curly-brace form, even for one-line if statements:

```suggestion
                        if (reindexMetrics != null) {
                            reindexMetrics.recordTookTime(elapsedTime);
                        }
```